gqcp/include/QCModel/HF/UHF.hpp

```diff
@@ -217,11 +232,11 @@ class UHF {
      * 
      *  @return the UHF direct (Coulomb) matrix for spin sigma
      */
-    static ScalarSQOneElectronOperator<Scalar> calculateScalarBasisDirectMatrix(const OneDM<Scalar>& P_alpha, const OneDM<Scalar>& P_beta, const SQHamiltonian<Scalar>& sq_hamiltonian) {
+    static ScalarSQOneElectronOperator<Scalar> calculateScalarBasisDirectMatrix(const SpinResolvedOneDM<Scalar>& P, const SQHamiltonian<Scalar>& sq_hamiltonian) {
```

Comment:
This method should return a `ScalarUSQoneElectronOperator`, but the other refactor is good!